src/serializers/serializer_io.c

```
@@ -165,7 +174,17 @@
-	return io->ReadBuffer(io->stream, lenptr);
+	size_t l;
+	char *v = io->ReadBuffer(io->stream, &l);
+
+	// save read buffer to memory stream
+	fwrite(v, sizeof(char), l, io->memstream);
+
+	if(lenptr != NULL) {
+		*lenptr = l;
+	}
+
+	return v;
```

cppcheck (2.10-2)

[error] Returning pointer to local variable 'l' that will be invalid when returning.

(returnDanglingLifetime)